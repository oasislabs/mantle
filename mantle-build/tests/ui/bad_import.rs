#[derive(mantle::Service)]
pub struct Service;

impl Service {
    pub fn new(_ctx: &mantle::Context) -> Self {
        Self
    }

    pub fn bad_import(&mut self, _ctx: &mantle::Context, arg: serde_cbor::Value) {}
}

fn main() {
    mantle::service!(Service);
}
