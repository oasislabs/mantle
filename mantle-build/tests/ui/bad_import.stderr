error: Cannot use types not defined in an RPC interface
 --> $DIR/bad_import.rs:9:63
  |
9 |     pub fn bad_import(&mut self, _ctx: &mantle::Context, arg: serde_cbor::Value) {}
  |                                                               ^^^^^^^^^^^^^^^^^

error: aborting due to previous error

